### PR TITLE
stream_info: add uint64 accessor

### PR DESCRIPTION
**Pull request by ohadvano**
Additional Description: None
Risk Level: Low
Testing: Unit tests
Docs Changes: None
Release Notes: None
Platform Specific Features: None

### CODE REVIEW

File: test/common/stream_info/uint64_accessor_impl_test.cc [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  UInt64AccessorImpl accessor(init_value);
+  accessor.increment();
+
+  EXPECT_EQ(0xdeadbeefdeadbef0, accessor.value());
+}
+
+TEST(UInt64AccessorImplTest, TestProto) {
+  UInt64AccessorImpl accessor(0xdeadbeefdeadbeef);
+  auto message = accessor.serializeAsProto();
+  EXPECT_NE(nullptr, message);
```
Comment (htuch) [inline test/common/stream_info/uint64_accessor_impl_test.cc:26]: Should validate the actual value.

File: envoy/stream_info/uint64_accessor.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+/**
+ * A FilterState object that tracks a single uint64_t value.
+ */
+class UInt64Accessor : public FilterState::Object {
+public:
+  /**
+   * Increments the tracked value by 1.
+   */
+  virtual void increment() PURE;
```
Comment (htuch) [inline envoy/stream_info/uint64_accessor.h:17]: Why increment and not some other forms of integer mutation?

Comment (ohadvano) [reply envoy/stream_info/uint64_accessor.h:17]: This PR is basically a copy of UInt32Accessor (in the same folder), but changed 32 to 64. I guess that if needed we could add more APIs in the future? The main use case for this class is saving an integer value in a filter state

File: test/common/stream_info/uint64_accessor_impl_test.cc (hunk 1 above)
Comment (ohadvano) [reply test/common/stream_info/uint64_accessor_impl_test.cc:27]: done

File: envoy/stream_info/uint64_accessor.h (hunk 2 above)
Comment (htuch) [reply envoy/stream_info/uint64_accessor.h:17]: Ah, fair.

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Pull request merged**